test/functional/autocmd/markset_spec.lua

```diff
@@ -0,0 +1,308 @@
+local t = require('test.testutil')
+local n = require('test.functional.testnvim')()
+
+local api = n.api
+local clear = n.clear
+local command = n.command
+local feed = n.feed
+local poke_eventloop = n.poke_eventloop
+local eval = n.eval
+
+local eq = t.eq
+local neq = t.neq
+
+describe('MarkSet', function()
+  -- TODO(justinmk): support other marks?: [, ] <, > . ^ " '
+
+  before_each(function()
+    clear()
+  end)
+
+  it('emits when lowercase/uppercase/[/] marks are set', function()
+    command([[
+      let g:mark_names = ''
+      let g:mark_events = []
+      autocmd MarkSet * call add(g:mark_events, {'event': deepcopy(v:event)}) | let g:mark_names ..= expand('<amatch>')
+      " TODO: there is a bug lurking here.
+      " autocmd MarkSet * let g:mark_names ..= expand('<amatch>')
```

Comment:
I think there's a bug lurking somewhere. If I uncomment this extra `MarkSet` handler, and remove the `let g:mark_names` from the above handler, then the `eq('ab', eval('g:mark_names'))` test fails.

This makes me wonder if there is a problem when more than one handler is defined.